modules/internal/ChapelReduce.chpl

```diff
@@ -213,6 +213,37 @@ module ChapelReduce {
     proc clone() return new unmanaged MinReduceScanOp(eltType=eltType);
   }
 
+  class minmax: ReduceScanOp {
+    type eltType;
+    var value = (max(eltType), min(eltType));
+
+    proc identity return (max(eltType), min(eltType));
+    proc accumulate(x){
+      if x.type == 2*eltType {
```

Comment:
Is this branch actually necessary?  What happens if it is not here?

Reply:
This branch is needed because reduce and scan are implemented slightly differently. While a reduction operation first uses ```accumulateOntoState``` and subsequently ```accumulate```, a scan operation does it in the opposite manner. So ```accumulate``` gets a tuple for reduction while ```accumulateOntoState``` does for scan.

As an example, the following are an instance of interleaved procedure calls for reduction and scan on the array ```[1,2,-3,43,5,66,7,-8,9,10]```

```chpl
// Reduction
Identity
Identity
Identity
accumulateOntoState 66 with state (9223372036854775807, -9223372036854775808)
accumulateOntoState 7 with state (66, 66)
accumulateOntoState -8 with state (7, 66)
accumulateOntoState 9 with state (-8, 66)
accumulateOntoState 10 with state (-8, 66)
accumulate (-8, 66) with local (9223372036854775807, -9223372036854775808)
combine {l = (l = false), value = (-8, 66)} with local (9223372036854775807, -9223372036854775808)
accumulateOntoState 1 with state (9223372036854775807, -9223372036854775808)
accumulateOntoState 2 with state (1, 1)
accumulateOntoState -3 with state (1, 2)
accumulateOntoState 43 with state (-3, 2)
accumulateOntoState 5 with state (-3, 43)
accumulate (-3, 43) with local (9223372036854775807, -9223372036854775808)
combine {l = (l = false), value = (-3, 43)} with local (-8, 66)
accumulate (9223372036854775807, -9223372036854775808) with local (-8, 66)


// Scan
accumulate 1 with local (9223372036854775807, -9223372036854775808)
accumulate 2 with local (1, 1)
accumulate -3 with local (1, 2)
accumulate 43 with local (-3, 2)
accumulate 66 with local (9223372036854775807, -9223372036854775808)
accumulate 5 with local (-3, 43)
accumulate 7 with local (66, 66)
accumulate -8 with local (7, 66)
accumulate 9 with local (-8, 66)
accumulate 10 with local (-8, 66)
Identity
accumulateOntoState (9223372036854775807, -9223372036854775808) with state (-3, 43)
accumulateOntoState (-3, 43) with state (-8, 66)
accumulateOntoState (9223372036854775807, -9223372036854775808) with state (1, 1)
accumulateOntoState (9223372036854775807, -9223372036854775808) with state (1, 2)
accumulateOntoState (9223372036854775807, -9223372036854775808) with state (-3, 2)
accumulateOntoState (-3, 43) with state (66, 66)
accumulateOntoState (9223372036854775807, -9223372036854775808) with state (-3, 43)
accumulateOntoState (9223372036854775807, -9223372036854775808) with state (-3, 43)
accumulateOntoState (-3, 43) with state (7, 66)
accumulateOntoState (-3, 43) with state (-8, 66)
accumulateOntoState (-3, 43) with state (-8, 66)
accumulateOntoState (-3, 43) with state (-8, 66)
```